src/api_server/src/request/machine_configuration.rs

```diff
@@ -65,25 +76,65 @@ mod tests {
     #[test]
     fn test_parse_put_machine_config_request() {
         assert!(parse_put_machine_config(&Body::new("invalid_payload")).is_err());
+        let mut body;
+        let mut expected_config: VmConfig;
+        #[cfg(target_arch = "x86_64")]
+        {
+            body = r#"{
+                "vcpu_count": 8,
+                "mem_size_mib": 1024,
+                "ht_enabled": true,
+                "cpu_template": "T2",
+                "track_dirty_pages": true
+              }"#;
 
-        let mut body = r#"{
+            expected_config = VmConfig {
+                vcpu_count: Some(8),
+                mem_size_mib: Some(1024),
+                ht_enabled: Some(true),
+                cpu_template: Some(CpuFeaturesTemplate::T2),
+                track_dirty_pages: true,
+            };
+
+            match vmm_action_from_request(parse_put_machine_config(&Body::new(body)).unwrap()) {
+                VmmAction::SetVmConfiguration(config) => assert_eq!(config, expected_config),
+                _ => panic!("Test failed."),
+            }
```

Comment:
This seems identical for x86_64 and arm. Should we put it outside the conditional blocks ?

Reply:
Sure .. will do that